### PR TITLE
add changes

### PR DESCRIPTION
**Pull request by SergiuLA17**

### CODE REVIEW

File: src/main/java/com/example/Storehouse/controllers/MainController.java [hunk 1]
```diff
@@ -1,39 +1,80 @@
...
-            storehouseService.updateProductWhenPurchased(productForUpdate, quantity);
-            try {
-                return ResponseEntity.ok(Product.toModel(product, quantity));
-            } catch (Exception e) {
-                throw new ProductNotfoundException();
-            }
-        } else
-            storehouseService.updateProductWhenProductNotFound(product);
-        throw new ProductNotfoundException();
+    SingleProductResponseBuilding singleProductResponse;
```
Comment (arthurdobrea) [inline src/main/java/com/example/Storehouse/controllers/MainController.java]: 1. Never inject RESPONSE OBJECT as class variable, thats bad practice. Instead whenever you need object of this type just use builder(and dont create classes with builder logic, instead use @builder annotstion.)

Comment (arthurdobrea) [reply src/main/java/com/example/Storehouse/controllers/MainController.java]: 2. Dont forget about final modificator

File: src/main/java/com/example/Storehouse/responseBuilder/multiproduct/MultiProductResponseBuilder.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import com.example.Storehouse.service.StorehouseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class MultiProductResponseBuilder implements iMultiProductResponseBuilder {
```
Comment (arthurdobrea) [inline src/main/java/com/example/Storehouse/responseBuilder/multiproduct/MultiProductResponseBuilder.java]: You can remove this class and migrate this to special Sevice class thst build all posible response classes

File: src/main/java/com/example/Storehouse/controllers/MainController.java (hunk 1 above)
Comment (SergiuLA17) [reply src/main/java/com/example/Storehouse/controllers/MainController.java]: done

File: src/main/java/com/example/Storehouse/responseBuilder/multiproduct/MultiProductResponseBuilder.java (hunk 2 above)
Comment (SergiuLA17) [reply src/main/java/com/example/Storehouse/responseBuilder/multiproduct/MultiProductResponseBuilder.java]: done

**Pull request closed without merging**